pkg/controller/metadata/metadata_controller.go

```diff
@@ -347,6 +349,47 @@ func (r *ReconcileMetadata) generateMetadataSecretData(instance *verifyv1beta1.M
 	return data, nil
 }
 
+// This function determines if we should regenerate the metadata or not.
+func ShouldRegenerate(secretsObj *corev1.Secret, hashOfRequestSpec string) bool {
+	secretsMap := secretsObj.Data
+
+	// If the rest of the config has changed and the hash has changed, regenerate regardless.
+	if secretsObj.ObjectMeta.Annotations[versionAnnotation] != hashOfRequestSpec {
+		return true
+	}
+
+	// Checking the date from the secrets store to see if we need to regenerate the metadata.
+	byteValidityDays := secretsMap[validityDays]
+	byteValidUntil := secretsMap[validUntil]
+
+	// If theres nothing in the map then regenerate metadata.
+	if byteValidityDays == nil || byteValidUntil == nil {
+		return true
+	}
+
+	intValidityDays, errValidityDays := strconv.Atoi(string(byteValidityDays))
+	validUntilTimeStamp, errValidUntil := time.Parse(time.RFC1123Z, string(byteValidUntil))
+
+	if errValidUntil != nil || errValidityDays != nil {
+		return true
+	}
+
+	// We want to regenerate the metadata if it's at least halfway through its lifetime.
+	var regeneratePastThisDate time.Time
+	if intValidityDays <= 1 {
+		regeneratePastThisDate = time.Now().Add(time.Duration(time.Hour * 12))
```

Comment:
https://github.com/alphagov/verify-metadata-controller/blob/7bb69eda78cea5ec156103d93693b3c4ac37e089/pkg/controller/metadata/metadata_controller.go#L379-L383

OK nice workaround! But it won't work for odd numbers.... We need to use hours for all values :)